google/cloud/ndb/model.py

```diff
@@ -4695,6 +4810,18 @@ class MyModel(ndb.Model):
     will create a query for the reserved ``__key__`` property.
     """
 
+    def __setstate__(self, state):
+        if type(state) is dict:
+            # this is not a legacy pb. set __dict__
+            self.__init__()
+            self.__dict__ = state
```

Comment:
I usually do `self.__dict__.update(state)`. I've never seen anyone replace __dict__ wholesale on an instance. Is that safe?

Reply:
I don't think this is all that risky, but I can update to use `update`